src/esp/sim/Simulator.cpp

```diff
@@ -148,9 +148,21 @@ void Simulator::reconfigure(const SimulatorConfiguration& cfg) {
 
     bool loadSuccess = false;
     if (config_.enablePhysics) {
-      loadSuccess = resourceManager_->loadScene(
-          sceneInfo, physicsManager_, &rootNode, &drawables,
-          cfg.sceneLightSetup, cfg.physicsConfigFile);
+      // use physics world attributes manager to get physics manager attributes
+      // described by config file
+      auto physicsManagerAttributes =
+          resourceManager_->getPhysicsAttributesManager()
+              ->createAttributesTemplate(cfg.physicsConfigFile, true);
```

Comment:
It looks like, deep inside `createAttributesTemplate` (specifically, inside `addTemplateToLibrary`), we construct a copy of the new attributes object. Can you explain what's going on here and why we need to make that copy?

Reply:
yes, all attributes are passed to the user as copies of what is kept in the library. No changes to the library occur unless attributes are specifically registered with the appropriate manager.  There are numerous reasons for this, such as allowing the user to reset the simulator while retaining copies of desired attributes, or allowing asset attribute edits to be validated before they are written to the library.

Reply:
Note : in this PR not all attributes getters send copies.  This is to support older functionality before this design decision was made to give users copies.  The full change-over to this mechanism will be completed in the next few PRs.